lnwallet/reservation.go

```diff
@@ -265,42 +265,66 @@ func (r *ChannelReservation) SetNumConfsRequired(numConfs uint16) {
+		return fmt.Errorf("csvDelay is too large: %d", csvDelay)
+	}
 
-	return nil
-}
+	// Fail if we consider the channel reserve to be too large.
+	// We currently fail if it is greater than 20% of the
+	// channel capacity.
+	maxChanReserve := r.partialState.Capacity / 5
+	if chanReserve > maxChanReserve {
+		return fmt.Errorf("chanReserve is too large: %g",
+			chanReserve.ToBTC())
+	}
 
-// RemoteChanConstraints returns our desired parameters which constraint the
-// type of commitment transactions that the remote party can extend for our
-// current state. In order to ensure that we only accept sane states, we'll
-// specify: the required reserve the remote party must uphold, the max value in
-// flight, and the maximum number of HTLC's that can propose in a state.
-func (r *ChannelReservation) RemoteChanConstraints() (btcutil.Amount, lnwire.MilliSatoshi, uint16) {
-	chanCapacity := r.partialState.Capacity
+	// Fail if the minimum HTLC value is too large. If this is
+	// too large, the channel won't be useful for sending small
+	// payments. This limit is currently set to maxValueInFlight,
+	// effictively letting the remote setting this as large as
+	// it wants.
+	// TODO(halseth): set a reasonable/dynamic value.
+	if minHtlc > maxValueInFlight {
+		return fmt.Errorf("minimum HTLC value is too large: %g",
+			r.ourContribution.MinHTLC.ToBTC())
+	}
 
-	// TODO(roasbeef): move csv delay calculation into func?
+	// Fail if maxHtlcs is above the maximum allowed number of 483.
+	// This number is specified in BOLT-02.
+	if maxHtlcs > uint16(MaxHTLCNumber/2) {
+		return fmt.Errorf("maxHtlcs is too large: %d", maxHtlcs)
+	}
 
-	// By default, we'll require them to maintain at least 1% of the total
-	// channel capacity at all times. This is the absolute amount the
-	// settled balance of the remote party must be above at *all* times.
-	chanReserve := (chanCapacity) / 100
+	// Fail if we consider maxHtlcs too small. If this is too small
+	// we cannot offer many HTLCs to the remote.
+	const minNumHtlc = 5
+	if maxHtlcs < minNumHtlc {
+		return fmt.Errorf("maxHtlcs is too small: %d", maxHtlcs)
+	}
 
-	// We'll allow them to fully utilize the full bandwidth of the channel,
-	// minus our required reserve.
-	maxValue := lnwire.NewMSatFromSatoshis(chanCapacity - chanReserve)
+	// Fail if we consider maxValueInFlight too small. We currently
+	// require the remote to at least allow minNumHtlc * minHtlc
+	// in flight.
+	if maxValueInFlight < minNumHtlc*minHtlc {
+		return fmt.Errorf("maxValueInFlight is too small: %g",
+			maxValueInFlight.ToBTC())
+	}
 
-	// Finally, we'll permit them to utilize the full channel bandwidth
-	maxHTLCs := uint16(MaxHTLCNumber / 2)
+	r.ourContribution.ChannelConfig.CsvDelay = csvDelay
+	r.ourContribution.ChannelConfig.ChanReserve = chanReserve
+	r.ourContribution.ChannelConfig.MaxAcceptedHtlcs = maxHtlcs
+	r.ourContribution.ChannelConfig.MaxPendingAmount = maxValueInFlight
+	r.ourContribution.ChannelConfig.MinHTLC = minHtlc
```

Comment:
This is incorrect. The minHTLC value they send is _theirs_, not what they're prescribing to us.

Reply:
But this means that this is a constraint we must be sure not to violate when sending them HTLCs? I think it makes it logical to group together with `MaxAcceptedHtlcs` and `MaxPendingAmount`.

(also see other comment)